packages/knip/src/plugins/expo/index.ts

```diff
@@ -0,0 +1,52 @@
+import type { IsPluginEnabled, Plugin, ResolveConfig, ResolveEntryPaths } from '../../types/config.js';
+import { toProductionEntry } from '../../util/input.js';
+import { join } from '../../util/path.js';
+import { hasDependency } from '../../util/plugin.js';
+import { getDependencies } from './helpers.js';
+import type { ExpoConfig } from './types.js';
+
+// https://docs.expo.dev/
+
+const title = 'Expo';
+
+const enablers = ['expo'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ['app.json', 'app.config.{ts,js}'];
+
+const resolveEntryPaths: ResolveEntryPaths<ExpoConfig> = async (expoConfig, { manifest }) => {
+  const config = 'expo' in expoConfig ? expoConfig.expo : expoConfig;
+
+  let production: string[] = ['index.{js,jsx,ts,tsx}'];
+
+  // https://docs.expo.dev/router/installation/#setup-entry-point
+  if (manifest.main === 'expo-router/entry') {
+    production = ['app/**/*.{js,jsx,ts,tsx}', 'src/app/**/*.{js,jsx,ts,tsx}'];
```

Comment:
Why does this `manifest.main` value result in all those files becoming entry files, isn't that pattern too broad?

Reply:
Setting `main` to `expo-router/entry` essentially "turns on" expo-router (if the package is installed, of course). It's a file-system based router, so every file in the `app` directory becomes it own route. (e.g. an `app/profile/settings.tsx` file will become a `/profile/settings` URL) This is all done under the hood by expo-router; these route files are not imported elsewhere in the project source code. So I do think the broad pattern is warranted here, you can name a file anything you want, and expo router will pick it up. It's the same concept as the NextJS "pages" router; anything in that directory is an entry file. https://github.com/webpro-nl/knip/blob/9ce4a49ab09b3e522fa48996f7f06cf44baee1c6/packages/knip/src/plugins/next/index.ts#L22

Reply:
Side note, the `app` directory can live either at the root or under `src`, and `src/app` takes precedence. (https://docs.expo.dev/router/reference/src-directory/) Is it worth it to check for the existence of `src/app` and fall back to `app` if it doesn't exist, instead of using both patterns at the same time?

Reply:
Sounds good, I'll keep it simple

Reply:
i also use expo-router but in a index.ts file as my entry and it seems that break that plugin, i can't get it to no pu everything as unused. maybe i configure it badly.

Reply:
Thanks @wcastand, here as a comment it might get lost, any chance you could give more details so we have something to look into? What's the issue you're seeing and how to reproduce it?

Reply:
i changed the way i handle the entry point so seems to be good for now, issue i can see seems to be with custom modules in `modules` directory being flagged at unused.
other than that hard to say what's an issue on our repo or not so will take a deeper look and if i find anything i will get back to you.

great tool either way thanks :)